tests/test_pipeline_mixin.py

```diff
@@ -685,3 +704,87 @@ def validate_test_components(test_case, task, model, tokenizer, processor):
             raise ValueError(
                 "Could not determine `vocab_size` from model configuration while `tokenizer` is not `None`."
             )
+
+
+def get_arg_names_from_hub_spec(hub_spec, first_level=True):
+    # This util is used in pipeline tests, to verify that a pipeline's documented arguments
+    # match the Hub specification for that task
+    arg_names = []
+    for field in fields(hub_spec):
+        # First, recurse into nested fields
+        if first_level and isclass(field.type) and issubclass(field.type, inference_specs.BaseInferenceType):
+            arg_names.extend(get_arg_names_from_hub_spec(field.type, first_level=False))
+            continue
+        # Next, catch nested fields that are part of a Union[], which is usually caused by Optional[]
+        for param_type in get_args(field.type):
+            if first_level and isclass(param_type) and issubclass(param_type, inference_specs.BaseInferenceType):
+                arg_names.extend(
+                    get_arg_names_from_hub_spec(param_type, first_level=False)
+                )  # Recurse into nested fields
+                break
+        else:
+            # Finally, this line triggers if it's not a nested field
+            arg_names.append(field.name)
+    return arg_names
+
+
+def parse_google_format_docstring_by_indentation(docstring):
+    # This util is used in pipeline tests, to extract the argument names from a google-format docstring
+    # to compare them against the Hub specification for that task
+    docstring = dedent(docstring)
+    lines_by_indent = [
+        (len(line) - len(line.lstrip()), line.strip()) for line in docstring.split("\n") if line.strip()
+    ]
+    args_lineno = None
+    args_indent = None
+    args_end = None
+    for lineno, (indent, line) in enumerate(lines_by_indent):
+        if line == "Args:":
+            args_lineno = lineno
+            args_indent = indent
+            continue
+        elif args_lineno is not None and indent == args_indent:
+            args_end = lineno
+            break
+    if args_lineno is None:
+        raise ValueError("No args block to parse!")
+    elif args_end is None:
+        args_block = lines_by_indent[args_lineno + 1 :]
+    else:
+        args_block = lines_by_indent[args_lineno + 1 : args_end]
+    outer_indent_level = min(line[0] for line in args_block)
+    outer_lines = [line for line in args_block if line[0] == outer_indent_level]
+    arg_names = [re.match(r"(\w+)\W", line[1]).group(1) for line in outer_lines]
+    return arg_names
+
+
+def compare_pipeline_args_to_hub_spec(pipeline_class, hub_spec):
+    docstring = inspect.getdoc(pipeline_class.__call__).strip()
+    docstring_args = set(parse_google_format_docstring_by_indentation(docstring))
+    js_args = set(get_arg_names_from_hub_spec(hub_spec))
```

Comment:
```suggestion
    hub_args = set(get_arg_names_from_hub_spec(hub_spec))
```

(nit) I would renamed them `hub_specs` as the method name suggests it (or `hf_specs` if you prefer). At the moment the specs as defined in `huggingface.js` but what makes them important is not that they are defined in javascript but that they are Hub-centric or HF-centric.